### PR TITLE
Fix a bug in parsing chunked messages

### PR DESCRIPTION
**Pull request by maskit**
This is for #10477 and #10580.

### CODE REVIEW

**Comment by maskit** [comment]
[approve ci autest]

File: src/proxy/http/HttpTunnel.cc [hunk 1]
```diff
@@ -157,8 +157,9 @@ ChunkedHandler::read_size()
           }
         } else {
           // We are done parsing size
-          if (num_digits == 0 || running_sum < 0) {
-            // Bogus chunk size
+          if ((num_digits == 0 || running_sum < 0) ||                               /* Bogus chunk size */
+              (!ParseRules::is_cr(*tmp) && !ParseRules::is_ws(*tmp) && *tmp != ';') /* Unexpected character */
+          ) {
```
Comment (ywkaras) [inline src/proxy/http/HttpTunnel.cc:162]: Taking a quick look at this, it seems like it would accept "100 junk  \n" as the chunk size of a 256 byte chunk.

Comment (maskit) [reply src/proxy/http/HttpTunnel.cc:162]: Yeah, a malformed chunk extension would pass the check. I ignored almost everything that follows a chunk size, but it should not allow reading random data as a chunk size.

Comment (bneradt) [reply src/proxy/http/HttpTunnel.cc:162]: Just making sure I understand: why is `;` an expected character? I don't ever recall seeing a `;` after the chunk header size.

Comment (bneradt) [reply src/proxy/http/HttpTunnel.cc:162]: Oh, should we also not be surprised by a LF? We seem to be lenient on that elsewhere in this loop.

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good in general. Just a couple questions to make sure I understand the patch.

File: src/proxy/http/HttpTunnel.cc (hunk 1 above)
Comment (maskit) [reply src/proxy/http/HttpTunnel.cc:162]: The semicolon is for `chunk-ext`.
```
  chunk          = chunk-size [ chunk-ext ] CRLF
                   chunk-data CRLF

  chunk-ext      = *( BWS ";" BWS chunk-ext-name
                      [ BWS "=" BWS chunk-ext-val ] )
```
https://www.rfc-editor.org/rfc/rfc9112#name-chunk-extensions

`BWS` is  Bad WhiteSpace, and it's basically optional whitespace.
https://www.rfc-editor.org/rfc/rfc9110.html#section-5.6.3-5

As for LF, I forgot to loosen the rule here. I pushed a commit to allow LF.

Comment (bneradt) [reply src/proxy/http/HttpTunnel.cc:162]: Sounds good. Thank you for the explanation.

**Review by bneradt — APPROVED** [review]
lgtm

**Comment by zwoop** [comment]
Cherry-picked to v9.2.x

**Pull request merged**